src/dev/impl/DevToys/ViewModels/Tools/Text/XmlValidator/XmlValidatorToolProvider.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Composition;
+using Windows.UI.Xaml.Controls;
+using DevToys.Api.Tools;
+using DevToys.Core.Threading;
+using DevToys.Shared.Api.Core;
+
+namespace DevToys.ViewModels.Tools.XmlValidator
+{
+    [Export(typeof(IToolProvider))]
+    [Name("Xml Validator")]
+    [Parent(TextGroupToolProvider.InternalName)]
+    [ProtocolName("xmlvalidator")]
+    [Order(1)]
+    [NotScrollable]
+    internal sealed class XmlValidatorToolProvider : ToolProviderBase, IToolProvider
+    {
+        private readonly IMefProvider _mefProvider;
+
+        public string MenuDisplayName => LanguageManager.Instance.XmlValidator.MenuDisplayName;
+
+        public string? SearchDisplayName => LanguageManager.Instance.XmlValidator.SearchDisplayName;
+
+        public string? Description => LanguageManager.Instance.XmlValidator.Description;
+
+        public string AccessibleName => LanguageManager.Instance.XmlValidator.AccessibleName;
+
+        public string? SearchKeywords => LanguageManager.Instance.XmlValidator.SearchKeywords;
+
+        public string? XmlData => LanguageManager.Instance.XmlValidator.XmlData;
+
+        public string? XsdScheme => LanguageManager.Instance.XmlValidator.XsdScheme;
+
+        public TaskCompletionNotifier<IconElement> IconSource => CreateSvgIcon("XmlValidator.svg");
+
+        [ImportingConstructor]
+        public XmlValidatorToolProvider(IMefProvider mefProvider)
+        {
+            _mefProvider = mefProvider;
+        }
+
+        public bool CanBeTreatedByTool(string data)
+        {
+            return false;
```

Comment:
What about supporting Smart Detection here. 😁
Here is an example:
https://github.com/veler/DevToys/blob/f55c5fb7d56e663cd92561fea219eee6d0ddf053/src/dev/impl/DevToys/ViewModels/Tools/Formatters/XmlFormatter/XmlFormatterToolProvider.cs#L40-L43

And then:
https://github.com/veler/DevToys/blob/f55c5fb7d56e663cd92561fea219eee6d0ddf053/src/dev/impl/DevToys/Views/Tools/Formatters/XmlFormatter/XmlFormatterToolPage.xaml.cs#L47-L50

Reply:
Should be supported now 😃